Source/websocket/WebLink.h

```diff
@@ -323,32 +323,29 @@ namespace Web {
         // -------------------------------------------------------------
         HAS_MEMBER(Transform, hasTransform);
 
-        typedef hasTransform<TRANSFORM, uint16_t (TRANSFORM::*)(BaseDeserializer&, uint8_t* data, const uint16_t maxSize)> TraitDeserializer;
-        typedef hasTransform<TRANSFORM, uint16_t (TRANSFORM::*)(BaseSerializer&, uint8_t* data, const uint16_t maxSize)> TraitSerializer;
-
-        template <typename CLASSNAME>
-        inline typename Core::TypeTraits::enable_if<CLASSNAME::TraitDeserializer::value, uint16_t>::type
-        ReceiveData(const CLASSNAME&, uint8_t* dataFrame, const uint16_t receivedSize)
+        template <typename CLASSNAME=TRANSFORM>
+        inline typename Core::TypeTraits::enable_if<hasTransform<CLASSNAME, uint16_t (CLASSNAME::*)(BaseDeserializer&, uint8_t* data, const uint16_t maxSize)>::value, uint16_t>::type
+        ReceiveData(uint8_t* dataFrame, const uint16_t receivedSize)
         {
             return (_transformer.Transform(_deserialiserImpl, dataFrame, receivedSize));
         }
 
-        template <typename CLASSNAME>
-        inline typename Core::TypeTraits::enable_if<!CLASSNAME::TraitDeserializer::value, uint16_t>::type
-        ReceiveData(const CLASSNAME&, uint8_t* dataFrame, const uint16_t receivedSize)
+        template <typename CLASSNAME= TRANSFORM>
+        inline typename Core::TypeTraits::enable_if<!hasTransform<CLASSNAME, uint16_t (CLASSNAME::*)(BaseDeserializer&, uint8_t* data, const uint16_t maxSize)>::value, uint16_t>::type
+        ReceiveData( uint8_t* dataFrame, const uint16_t receivedSize)
         {
             return (_deserialiserImpl.Deserialize(dataFrame, receivedSize));
         }
 
-        template <typename CLASSNAME>
-        inline typename Core::TypeTraits::enable_if<CLASSNAME::TraitSerializer::value, uint16_t>::type
+        template <typename CLASSNAME=TRANSFORM>
+        inline typename Core::TypeTraits::enable_if<hasTransform<CLASSNAME, uint16_t (CLASSNAME::*)(BaseSerializer&, uint8_t* data, const uint16_t maxSize)>::value, uint16_t>::type
         SendData(const CLASSNAME&, uint8_t* dataFrame, const uint16_t receivedSize)
```

Comment:
Think the first param can be removed as well now.

Reply:
removed it a s well from the has_Transform .